### PR TITLE
Add the option to disable Thalia pay for events

### LINKED ISSUE #1422: Disable Thalia Pay payments for specific events
### Is your feature request related to a problem? Please describe.
There might be events where Thalia Pay should not be available (study trip, events where the payment amount is variable like the orientation bbq, events where registration is not first come first serve like wine tasting or committee lunch)

### Describe the solution you'd like
An option in the events admin that sets whether Thalia Pay is allowed for this event. The default should be true. When false, the payment button should not appear.

Probably we should also add something to the Payable interface about this so we can implement the same check in the process_payment code.

### Motivation
Thalia Pay is quite critical and should work as intended always without too much manual hassle.

### Describe alternatives you've considered
Manually removing payments that we do not want.

### PR DESCRIPTION
**Pull request by 9Y0**
Closes #1422 

### Summary
You can now disable Thalia Pay for specififc events

### How to test
1. Create an event with a price and uncheck `Is Thalia Pay payment allowed for this event`
2. Make sure you have Thalia Pay enabled
3. Notice that there is no longer a `Pay with Thalia Pay` button on the event page

### CODE REVIEW

File: website/events/models/event.py [hunk 1]
```diff
@@ -179,6 +179,10 @@ class Event(models.Model, metaclass=ModelTranslateMeta):
         null=True,
     )
 
+    tpay_allowed = models.BooleanField(
+        _("Is Thalia Pay payment allowed for this event"), default=True, blank=False
```
Comment (se-bastiaan) [inline website/events/models/event.py:183]: Boolean fields can never be blank. They are either False or True.

**Comment by se-bastiaan** [comment]
deploy review

**Comment by se-bastiaan** [comment]
remove deploy

File: website/events/models/event.py [hunk 2]
```diff
@@ -179,6 +179,10 @@ class Event(models.Model, metaclass=ModelTranslateMeta):
         null=True,
     )
 
+    tpay_allowed = models.BooleanField(
+        _("Is Thalia Pay payment allowed for this event"), default=True
```
Comment (se-bastiaan) [inline website/events/models/event.py:183]: image
This text looks a bit weird :p

**Review by se-bastiaan — CHANGES_REQUESTED** [review]
Code looks good! Maybe discuss the text of the checkbox?

File: website/events/models/event.py (hunk 2 above)
Comment (9Y0) [reply website/events/models/event.py:183]: Yeah that indeed looks weird.
I think `Allow Thalia Pay payments` would be better. We could also go for `Allow Thalia Payments` but I'm not sure if we want to use that terminology (maybe @JobDoesburg can answer this)

Comment (se-bastiaan) [reply website/events/models/event.py:183]: I think the first one is best. Since 'Thalia Pay' is kind of the 'thing'.

Comment (se-bastiaan) [reply website/events/models/event.py:183]: Does that make sense? I don't know :p

Comment (JobDoesburg) [reply website/events/models/event.py:183]: Or just "Allow Thalia Pay"

**Review by se-bastiaan — APPROVED** [review]

**Review by JobDoesburg — APPROVED** [review]

**Pull request merged**